src/js/angular/models/import/import-resource-tree-element.js

```diff
@@ -137,4 +137,54 @@ export class ImportResourceTreeElement {
         }
         return null;
     }
+
+    hasTextInResourcesName(searchText) {
+        if (this.importResource.name.includes(searchText)) {
+            return true;
+        }
+        if (this.files.some((file) => file.importResource.name.includes(searchText))) {
+            return true;
+        }
+        for (const directory of this.directories) {
```

Comment:
What's wrong with the `return this.directories.some((directory) => directory.hasTextInResourcesName(searchText))`? And same for the next two methods